vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/metis_items.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {useEffect} from 'react';
+import {WithInputParams} from './input_types';
+import {some, withDefault} from '../../../../selectors/maybe';
+import { PickFileOrFolder, PickBucket } from 'etna-js/components/metis_exploration';
+import {useSetsDefault} from './useSetsDefault';
+import { Grid } from '@material-ui/core';
+
+declare const CONFIG: {[key: string]: any};
+type metisPathType = { bucket: string, path: string, type: 'file' | 'folder' | null }
+
+function _MetisLocationInput({onChange, label, allowFiles, data, ...props}: WithInputParams<{
+    label?: string,
+    allowFiles: boolean
+    }, metisPathType, metisPathType
+  >) {
+  const value = useSetsDefault({bucket: '', path: '', type: null}, props.value, onChange);
+
+  function updateKeyToVal(key: 'bucket' | 'path' | 'type', val: string | null, fullValues = {...value}) {
+    // console.log('setting ', key, ' to ', val)
+    const newValue = fullValues
+    if (newValue[key] != val) {
+      // only push when different than current
+      newValue[key] = val as any
+      onChange(some(newValue))
+    }
+  }
+
+  function updateType(t: 'file' | 'folder' | null) {
+    updateKeyToVal('type', t)
+  }
+
+  useEffect( () => {
+    if (value.bucket == '' && value.type != null) {
+      updateType(null)
+    } 
+  }, [value.bucket])
+
+  return (
+    !value || value.bucket == undefined ? null :
+    <Grid container direction='column'>
+      <Grid item>
+        <PickBucket
+          setBucket={ (b: string) => {
+            updateKeyToVal('bucket', b)
+          }}
+          bucket={value.bucket}
+          label={label? label+', '+'Bucket' : undefined}
+          disablePortal={false}
+        />
+      </Grid>
+      <Grid>
+        <PickFileOrFolder
```

Comment:
Think it doesn't filter non-CSV/TSV files yet.

Reply:
Right!  It doesn't.  Had meant to give an error via the python script that reads in the file, but forgot.  Not the best way anyway.

I wonder if there's a way to pass in a path regex from the CWL so that we don't have to make new input_type for all unique file typings we might want...

For now, I'll add the single one for this csv/tsv case & the check machinery via the validator, before merge.